src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -87,7 +104,9 @@ public void deleteAssembly(AssemblyEntity entity) {
 
     public boolean isEntityPresent(AssemblyEntity entity) {
         Optional<AssemblyEntity> existingAssembly = repository.findAssemblyEntityByGenbankOrRefseq(
-                entity.getGenbank(), entity.getRefseq());
+                // Setting to invalid prevents finding random accessions with null GCA/GCF
```

Comment:
Earlier different assemblies were getting matched if they both had `refget == null`. Now I am substituing `null` with an "invalid" pattern so only genuine entries get found. The alternative to this would be writing a custom query.